internal/services/springcloud/spring_cloud_dynatrace_application_performance_monitoring_resource.go

```diff
@@ -0,0 +1,348 @@
+package springcloud
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/appplatform/2023-09-01-preview/appplatform"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type SpringCloudDynatraceApplicationPerformanceMonitoringModel struct {
+	Name                 string `tfschema:"name"`
+	SpringCloudServiceId string `tfschema:"spring_cloud_service_id"`
+	GloballyEnabled      bool   `tfschema:"globally_enabled"`
+	ApiUrl               string `tfschema:"api_url"`
+	ApiToken             string `tfschema:"api_token"`
+	ConnectionPoint      string `tfschema:"connection_point"`
+	EnvironmentId        string `tfschema:"environment_id"`
+	Tenant               string `tfschema:"tenant"`
+	TenantToken          string `tfschema:"tenant_token"`
+}
+
+type SpringCloudDynatraceApplicationPerformanceMonitoringResource struct{}
+
+var _ sdk.ResourceWithUpdate = SpringCloudDynatraceApplicationPerformanceMonitoringResource{}
+
+func (s SpringCloudDynatraceApplicationPerformanceMonitoringResource) ResourceType() string {
+	return "azurerm_spring_cloud_dynatrace_application_performance_monitoring"
+}
+
+func (s SpringCloudDynatraceApplicationPerformanceMonitoringResource) ModelObject() interface{} {
+	return &SpringCloudDynatraceApplicationPerformanceMonitoringModel{}
+}
+
+func (s SpringCloudDynatraceApplicationPerformanceMonitoringResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return appplatform.ValidateApmID
+}
+
+func (s SpringCloudDynatraceApplicationPerformanceMonitoringResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"spring_cloud_service_id": commonschema.ResourceIDReferenceRequiredForceNew(commonids.SpringCloudServiceId{}),
+
+		"connection_point": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"tenant": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			Sensitive:    true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"tenant_token": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			Sensitive:    true,
+			ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
and here

Reply:
This is the token used to authenticated, so we can't validate it.